src/common/proc_macro/src/guc.rs

```diff
@@ -0,0 +1,239 @@
+pub struct GucOpts {
+    pub rename: Option<syn::LitStr>,
+    pub default: syn::Expr,
+    pub flags: Option<syn::LitStr>,
+    pub check_hook: Option<syn::Expr>,
+}
+
+pub(crate) fn derive_guc(input: DeriveInput) -> TokenStream {
+    let syn::Data::Struct(syn::DataStruct { fields, .. }) = input.data else {
+        abort!(input, "Only struct is supported");
+    };
+
+    let mut default_fields = vec![];
+    let mut struct_impl_set = vec![];
+    let mut struct_impl_get = vec![];
+    let mut struct_impl_reset = vec![];
+    let mut set_match_branches = vec![];
+    let mut get_match_branches = vec![];
+    let mut show_all_list = vec![];
+
+    for field in fields {
+        let field_ident = field.ident.expect_or_abort("Field need to be named");
+        let ty = field.ty;
+
+        let mut doc_list = vec![];
+        for attr in &field.attrs {
+            if attr.path.is_ident("doc") {
+                let meta = attr.parse_meta().expect_or_abort("Failed to parse meta");
+                if let syn::Meta::NameValue(val) = meta {
+                    if let syn::Lit::Str(desc) = val.lit {
+                        doc_list.push(desc.value().trim().to_string());
+                    }
+                }
+            }
+        }
+
+        let description: TokenStream = format!("r#\"{}\"#", doc_list.join(" ")).parse().unwrap();
+
+        let attr =
+            GucOpts::from_attributes(&field.attrs).expect_or_abort("Failed to parse attribute");
+        let GucOpts {
+            rename,
+            default,
+            flags,
+            check_hook: check_hook_name,
+        } = attr;
+
+        let entry_name = if let Some(rename) = rename {
+            if !(rename.value().is_ascii() && rename.value().to_ascii_uppercase() == rename.value())
+            {
+                abort!(rename, "Expect `rename` to be an ascii upper case string");
+            }
+            quote! {#rename}
+        } else {
+            let ident = format_ident!("{}", field_ident.to_string().to_uppercase());
+            quote! {stringify!(#ident)}
+        };
+
+        let flags = flags.map(|f| f.value()).unwrap_or_default();
+        let flags: Vec<_> = flags.split('|').map(|str| str.trim()).collect();
+
+        default_fields.push(quote_spanned! {
+            field_ident.span()=>
+            #field_ident: #default.into(),
+        });
+
+        let set_func_name = format_ident!("set_{}_str", field_ident);
+        let set_t_func_name = format_ident!("set_{}", field_ident);
+        let set_t_inner_func_name = format_ident!("set_{}_inner", field_ident);
+
+        let gen_set_func_name = if flags.contains(&"SETTER") {
+            set_t_inner_func_name.clone()
+        } else {
+            set_t_func_name.clone()
+        };
+
+        let check_hook = if let Some(check_hook_name) = check_hook_name {
+            quote! {
+                #check_hook_name(&val).map_err(|_e| {
+                    ErrorCode::InvalidConfigValue {
```

Comment:
It's a good practice to always generated qualified path in the proc macro. For example, we cannot neither assume `ErrorCode` is imported into the current scope nor assume it's exactly `risingwave_common::error::ErrorCode`. So theoretically, we should write `::risingwave_common::error::ErrorCode` here. Same for `Result`.